cmd/broker/main.go

```diff
@@ -300,22 +299,19 @@ func main() {
 		fatalOnError(err, log)
 	}
 
-	regionsSupportingMachine, err := regionssupportingmachine.ReadRegionsSupportingMachineFromFile(cfg.RegionsSupportingMachineFilePath)
-	fatalOnError(err, log)
-	log.Info(fmt.Sprintf("Number of machine types families that are not universally supported across all regions: %d", len(regionsSupportingMachine)))
-
-	workersProvider := workers.NewProvider(cfg.InfrastructureManager, regionsSupportingMachine)
-
 	plansConfiguration, err := os.Open(cfg.PlansConfigurationFilePath)
 	fatalOnError(err, log)
 	plansSpec, err := configuration.NewPlanSpecificationsFromFile(cfg.PlansConfigurationFilePath)
 	fatalOnError(err, log)
 	providersConfiguration, err := os.Open(cfg.ProvidersConfigurationFilePath)
 	fatalOnError(err, log)
+	providerSpec, err := configuration.NewProviderSpecFromFile(cfg.ProvidersConfigurationFilePath)
+	fatalOnError(err, log)
```

Comment:
Lots of direct/indirect file opening funcs. I think you can pass `plansSpec` and `providerSpec` to `NewSchemaService` instead files descriptors.